CDCgov/prime-reportstream
# Generate SPM segments from FHIR

This PR ...
- Adds the generation of the SPM segment
- Adds a "coded with exceptions" CWE HL7 mapping helper.

Test Steps:
1. Run the FHIR -> HL7 conversion command ```./gradlew primeCLI --args='fhirdata --input-file "src/testIntegration/resources/datatests/HL7_to_FHIR/sample_co_1_20220518-0001.fhir" -s metadata/hl7_mapping/ORU_R01/ORU_R01-base.yml'```
2. You will notice these differences:
- SPM.2.2.4 Doesn't include ISO . This follows the previous decision not to include ISO if the universal id is not included
- SPM.14 only includes RA, This is noted in the mapping and is due to the fact that HCA is supposed to be sending us text but instead is sending a CE
- SPM.26. In the sample file some numbers are ints while others are doubles. The output of the generated HL7 is always a double.

## Changes
- Adds the generation of the SPM segment
- Adds a "coded with exceptions" CWE HL7 mapping helper.
- Adds a None concept to the CodingSystemMapper so throw an exception if there is no mapping.

## Checklist

### Testing

## Linked Issues
- #5755
